### PR TITLE
Handle errors in additionalItemBackupper

### LINKED ISSUE #509: Backups with errors aren't marked as partially complete
I had a permission issue with my cloud credentials which resulted in Ark not being able to snapshot volumes. However, without going through the logs, it seems like there is no indication of this failure: `ark backup get` lists it as ´Completed`, and `ark backup describe my-backup` only has a field for `Validation errors`, which is empty (reasonably enough).

Shouldn't there be some more indication that errors happened at least? Since I didn't see any errors, I restored the backup, and get to a semi-broken state since the pv/pvc objects aren't created.

### PR DESCRIPTION
**Pull request by carlpett**
Fixes #509.

Currently the corresponding test fails, which I _think_ is because of #511.

### CODE REVIEW

File: pkg/backup/item_backupper_test.go [hunk 1]
```diff
@@ -317,11 +329,16 @@ func TestBackupItemNoSkips(t *testing.T) {
 
 			var snapshotService *arktest.FakeSnapshotService
 			if test.snapshottableVolumes != nil {
-				snapshotService = &arktest.FakeSnapshotService{
-					SnapshottableVolumes: test.snapshottableVolumes,
-					VolumeID:             "vol-abc123",
+				if test.snapshotError != nil {
```
Comment (carlpett) [inline pkg/backup/item_backupper_test.go]: This is a bit of a mess - would appreciate some ideas on how to make this clearer

File: pkg/util/test/fake_snapshot_service.go [hunk 2]
```diff
@@ -89,3 +89,31 @@ func (s *FakeSnapshotService) SetVolumeID(pv runtime.Unstructured, volumeID stri
 	s.VolumeIDSet = volumeID
 	return pv, nil
 }
+
```
Comment (carlpett) [inline pkg/util/test/fake_snapshot_service.go]: Not sure if this is the right place?

File: pkg/backup/item_backupper.go [hunk 3]
```diff
@@ -206,7 +206,10 @@ func (ib *defaultItemBackupper) backupItem(logger logrus.FieldLogger, obj runtim
 					return err
 				}
 
-				ib.additionalItemBackupper.backupItem(log, additionalItem, gvr.GroupResource())
+				err = ib.additionalItemBackupper.backupItem(log, additionalItem, gvr.GroupResource())
```
Comment (ncdc) [inline pkg/backup/item_backupper.go]: When we are only checking the returned error and aren't doing anything else with it, we usually use the syntax `if err := foo(); err != nil {`

File: pkg/util/test/fake_snapshot_service.go (hunk 2 above)
Comment (ncdc) [reply pkg/util/test/fake_snapshot_service.go]: Instead of creating a new type, please add `err error` to `FakeSnapshotService` and return `s.err` wherever we need to return an error.

File: pkg/backup/item_backupper_test.go (hunk 1 above)
Comment (ncdc) [reply pkg/backup/item_backupper_test.go]: See my comment below

File: pkg/backup/item_backupper.go (hunk 3 above)
Comment (carlpett) [reply pkg/backup/item_backupper.go]: Right. I'll do the same for the three if:s above at the same time, then?

Comment (carlpett) [reply pkg/backup/item_backupper.go]: Ah, sorry. That doesn't work though, since they set other vars. Never mind, then, just fixing this one :)

File: pkg/backup/item_backupper_test.go [hunk 4]
```diff
@@ -242,6 +243,17 @@ func TestBackupItemNoSkips(t *testing.T) {
 				"vol-abc123": {SnapshotID: "snapshot-1", AvailabilityZone: "us-east-1c"},
 			},
 		},
+		{
+			name: "backup fails when takePVSnapshot fails",
```
Comment (ncdc) [inline pkg/backup/item_backupper_test.go]: To test your change to item_backupper.go, we need a test case that has this mocked behavior return a non-nil error: `additionalItemBackupper.On("backupItem", mock.AnythingOfType("*logrus.Entry"), test.customActionAdditionalItems[i], item.GroupResource).Return(nil)`.

The test case you've added here is valuable too, so we should keep it, but it doesn't cover your fix: namely, that when we call `ib.additionalItemBackupper.backupItem(log, additionalItem, gvr.GroupResource())`, it checks the returned error and returns it if it's non-nil.

Comment (ncdc) [reply pkg/backup/item_backupper_test.go]: Something like this would cover your change: https://gist.github.com/ncdc/12c3b6b6f4db0e5c60f4ecf77e8d85c2

Comment (carlpett) [reply pkg/backup/item_backupper_test.go]: Thanks! Updated with this, will ask some questions on Slack too :)

**Comment by ncdc** [comment]
LGTM. @skriss ptal
